File: tests/fixtures/typescript/sample.ts
const MAX_TIMEOUT: number = 5000;

interface User {
    id: number;
    name: string;
}

interface Searchable {
    search(query: string): User[];
}

class BaseService {
    protected log(msg: string): void {}
}

class UserService extends BaseService implements Searchable {
    getUser(userId: number): User {
        return { id: userId, name: "" };
    }

    search(query: string): User[] {
        return [];
    }
}

function authenticate(token: string): boolean {
    return token.length > 0;
}

type UserID = number;

File: tests/fixtures/lua/sample.lua
-- Maximum number of retries
MAX_RETRIES = 3

-- Authenticate a token
function authenticate(token)
    return token ~= nil and #token > 0
end

-- User service module
local UserService = {}

function UserService.getUser(userId)
    return { id = userId, name = "test" }
end

return UserService
